packages/fxa-settings/src/components/UnitRowTwoStepAuth/index.tsx

```diff
@@ -0,0 +1,114 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import React, { useState } from 'react';
+import { gql, useMutation } from '@apollo/client';
+import LinkExternal from 'fxa-react/components/LinkExternal';
+import { useBooleanState } from 'fxa-react/lib/hooks';
+import AlertBar from '../AlertBar';
+import Modal from '../Modal';
+import UnitRow from '../UnitRow';
+import VerifiedSessionGuard from '../VerifiedSessionGuard';
+import { useAccount } from '../../models';
+
+export const DESTROY_TOTP_TOKEN_MUTATION = gql`
+  mutation destroyTotpToken($input: DestroyTotpToken!) {
+    destroyTotpToken(input: $input) {
+      clientMutationId
+    }
+  }
+`;
+
+export const UnitRowTwoStepAuth = () => {
+  const { totp } = useAccount();
+  const { exists, verified } = totp;
+  const [modalRevealed, revealModal, hideModal] = useBooleanState();
+  const [alertBarRevealed, revealAlertBar, hideAlertBar] = useBooleanState();
+  const [errorText, setErrorText] = useState<string>();
+  const onError = (e: Error) => {
+    setErrorText(e.message);
+    hideModal();
+    revealAlertBar();
+  };
+  const [disableTwoStepAuth] = useMutation(DESTROY_TOTP_TOKEN_MUTATION, {
+    variables: { input: {} },
+    onCompleted: () => {
+      hideModal();
+      revealAlertBar();
+    },
+    onError,
+    ignoreResults: true,
+    update: (cache) => {
+      cache.modify({
+        fields: {
+          account: (existing) => {
+            return { ...existing, totp: false };
+          },
+        },
+      });
+    },
+  });
+
+  return (
+    <UnitRow
+      header="Two-step authentication"
+      headerValueClassName={exists ? 'text-green-800' : ''}
+      headerValue={exists ? 'Enabled' : 'Not Set'}
```

Comment:
I think we can simplify this a tad since `noHeaderValueText` is also a prop here and there's a lot of conditionals based on `exists`. What do you think about something like:

```
  const conditionalUnitRowProps = exists ? {
    headerValueClassName: 'text-green-800',
    headerValue: 'Enabled',
    secondaryButtonClassName: 'bg-red-500 text-white border-red-600',
    revealSecondaryModal: revealModal,
  } : {
    headerValue: null,
    noHeaderValueText: 'Not Set',
    secondaryCtaText: undefined,
    revealSecondaryModal: undefined
  };

  return (
    <UnitRow
      header="Two-step authentication"
      route="/beta/settings/two_step_authentication"
      {...conditionalUnitRowProps}
    >
```